### PR TITLE
2063 rdctl shell

### LINKED ISSUE #2063: Add `rdctl shell` command
On Linux and macOS, running `rdctl shell` is the same as:
```
export LIMA_HOME=<path to lima dir, (currently) parent of 0 and _config>
export PATH=<path to resources/PLATFORM/lima/bin>
limactl shell 0
```

If args are passed to `rdctl shell`, they're passed on to `limactl shell 0`. If an arg `'abc def'` is quoted like here, the quotedness is passed on to `limactl`. Note that this might require a `--` after the `rdctl` flags to prevent `rdctl` from processing them.

On Windows, running `rdctl shell` is the same as running `wsl`, and until we find a reason to do otherwise,
it takes the same arguments as `wsl`. In effect, it's the opposite of an abbreviation alias of `wsl`. Same caveat that if the first wsl argument starts with a `-`, it will most likely need to be preceded by a `--` argument to avoid being processed by `rdctl`.

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #2063

### CODE REVIEW

**Comment by ericpromislow** [comment]
Waiting for PR #2061 to be merged.

File: src/go/rdctl/cmd/shell.go
Comment (jandubois) [inline src/go/rdctl/cmd/shell.go:66]: This sounds wrong; we don't want to use any `limactl` from the `PATH`, but always the version bundled with Rancher Desktop.

File: src/go/rdctl/cmd/shell.go
Comment (jandubois) [inline src/go/rdctl/cmd/shell.go:71]: This can just be:

```suggestion
		commandName = path.Join(path.Dir(path.Dir(execPath)), "lima", "bin", "limactl")
```

There is no need to check the access mode etc; we bundle this binary. If it fails, just display the "path not found" or whatever happens when you run it.

File: src/go/rdctl/cmd/shell.go
Comment (jandubois) [inline src/go/rdctl/cmd/shell.go:112]: Again, we don't want this. We only ever want to use the Rancher Desktop VM. Checking the environment will break `rdctl shell` because it will point to the user's `LIMA_HOME`, which will not contain the `0` Rancher Desktop instance.

File: src/go/rdctl/cmd/root.go
Comment (jandubois) [inline src/go/rdctl/cmd/root.go:83]: Why are all the other cases needed? Why isn't this just:

```suggestion
	if len(os.Args) > 1 && os.Args[0] == "shell" {
		return
	}
```

Please add a comment why config parsing must be disabled for all the other variants as well!

**Review by jandubois — CHANGES_REQUESTED** [review]
The code that looks up `limactl` on the `PATH` or uses `LIMA_HOME` from the current environment must be removed. The embedded Lima setup must be treated as completely separate from the user's Lima configuration, so neither should affect the other.

**Comment by jandubois** [comment]
I've added a commit making the changes to `cmd/shell.go` that I would want to see before I can approve this PR.

I haven't made any changes to `cmd/root.go`, as I still don't understand why the complex condition is required. I'm fine with leaving it as-is for now and fixing it later, as I don't think it will create any issues; it just looks unnecessary.

**Review by jandubois — APPROVED** [review]
Approving now to still get it into the release; the remaining comments can be addressed in a followup PR.

File: src/go/rdctl/cmd/root.go
Comment (ericpromislow) [reply src/go/rdctl/cmd/root.go:83]: I'll add this comment in a PR post-release:

The `shell`, `version`, and `completion` commands have no use for the config or any of the config-related options
And if they're running something like `rdctl help shell | version | completion` we don't want to show the config-related options in help.

I didn't find any info about building artificial hierarchies with cobra.  A "real" hierarchy would require another subcommand, as in `rdctl no-config shell`or `rdctl config api OPTIONS`  but we don't want to do that.

**Pull request merged**
